cpp/src/arrow/flight/sql/client.h

```diff
@@ -364,25 +364,15 @@ class ARROW_FLIGHT_SQL_EXPORT FlightSqlClient {
   Status Close();
 
  protected:
```

Comment:
@lidavidm would it be a problem to make these method `public`? It would solve a problem I'm having and it would enable people to also be able to extend`` the `FlightSqlClient` by composition and not just inheritance.

Reply:
These are here basically only for testing, so that GMock can mock them out for something else. What is the problem?

Reply:
I want to create a wrapper/helper that can't live in the header (as private method) but it needs to call `DoPut` and `DoGet`.

Additionally: given that these are just for testing and FlightClient is injectable I think a level of indirection can be removed by mocking the `FlightClient` instead of the `FlightSqlClient`.

Reply:
In C++ that requires declaring everything as `virtual`, though, and FlightClient wasn't really meant as an interface. (Perhaps that could be changed.)

Reply:
> and FlightClient wasn't really meant as an interface

Oh, right. I will figure something. At the moment it's very confusing that some of the wrapper methods are virtual while others aren't.